### DISCUSSION TITLE
Arch Linux and Dbus mediation

### DISCUSSION THREAD

**Discussion by curiosityseeker** [roddhjav/apparmor.d#266] [General]
Arch Linux is [planning](https://gitlab.archlinux.org/archlinux/rfcs/-/merge_requests/25) to make dbus-broker its default dbus daemon. The reasoning is that [dbus-broker provides better performance and higher reliability than dbus-daemon, with per-user accounting of resources in the broker](https://gitlab.archlinux.org/heftig/rfcs/-/blob/dbus-broker/rfcs/0025-dbus-broker-default.rst).

However, it is [argued](https://gitlab.archlinux.org/archlinux/rfcs/-/merge_requests/25#note_147976) that dbus-broker has no AppArmor support (also [mentioned](https://wiki.archlinux.org/title/D-Bus) in the Arch wiki).

On the other hand, it's mentioned [here](https://apparmor.pujol.io/issues.html#gnome-can-be-very-slow-to-start) that "on archlinux, the dbus mediation is not enabled". Why is that? Is that caused by the fact that only the kernels used in Ubuntu have a respective [patch](https://gitlab.archlinux.org/archlinux/rfcs/-/merge_requests/25#note_148291) added which hasn't landed yet in the mainstream kernel? If so - wouldn't that mean that at the time being it doesn't make a difference for AppArmor if dbus-broker or dbus-daemon is used in Arch Linux?

**Comment by roddhjav** [comment]
As mentioned in the comment on arch's Gitlab: AppArmor integration with dbus is an Ubuntu feature requiring a patched kernel adding `SO_PEERSEC` **and** a patched dbus-daemon. So as you said, arch decision would not change anything anyway.

Now, I am aware that the ubuntu kernel patch is going in to merged upstream, at this point, we could consider what is the best option (adding apparmor support to dbus-broker most likely)

**Comment by curiosityseeker** [comment]
Thanks! So my interpretation was correct and we have to wait until those patches will be upstreamed. [This](https://github.com/bus1/dbus-broker/pull/286) is the PR for adding AppArmor support to dbus-broker.

**Comment by curiosityseeker** [comment]
FWIW, the transition to dbus-broker [has taken place](https://archlinux.org/news/making-dbus-broker-our-default-d-bus-daemon/). 2 processes - `dbus-broker` and `dbus-broker-launch` - need to be confined.
